webapp/src/components/Modal/BidConfirmationModal/BidConfirmationModal.js

```diff
@@ -24,6 +32,60 @@ export default class BidConfirmationModal extends React.PureComponent {
     onAuthorize: PropTypes.func
   }
 
+  constructor(props) {
+    super(props)
+    this.state = {
+      isLoading: false,
+      balance: null
+    }
+  }
+
+  componentWillMount() {
+    const { token, address } = this.props
+    this.fetchBalance(token, address)
+  }
+
+  componentWillReceiveProps({ token, address }) {
+    if (this.props.token !== token || this.props.address !== address) {
+      this.fetchBalance(token, address)
+    }
+  }
+
+  fetchBalance = async (symbol, address) => {
+    this.setState({ balance: null, isLoading: true })
+
+    try {
+      const contractAddress = TOKEN_ADDRESSES[symbol]
+      const response = await axios.get(
+        `https://api.etherscan.io/api?module=account&action=tokenbalance&contractaddress=${contractAddress}&address=${address}&tag=latest&apikey=${env.get(
+          'REACT_APP_ETHERSCAN_API_KEY'
+        )}`
+      )
+      const balance =
+        Number(response.data.result) / (symbol === 'ZIL' ? 10 ** 12 : 10 ** 18)
```

Comment:
We are using `parseInt(value, 10)` in every places. Maybe we could use it here too

Reply:
done 👍